src/sentry/features/base.py

```diff
@@ -1,24 +1,40 @@
 __all__ = ["Feature", "OrganizationFeature", "ProjectFeature", "ProjectPluginFeature"]
 
+from abc import ABC
+from typing import TYPE_CHECKING, Any
 
-class Feature:
-    def __init__(self, name):
+if TYPE_CHECKING:
+    from sentry.models import Organization, Project
+
+
+class Feature(ABC):
+    def __init__(self, name: str, *args: Any, **kwargs: Any) -> None:
+        """
+        `FeatureManager.get()` and `FeatureCheckBatch.get_feature_objects()`
+        expect to be able to pass a `Feature` arbitrary `args` and `kwargs`.
+        """
         self.name = name
 
+    def get_organization(self) -> "Organization":
```

Comment:
I think you need to decorate this as abstract method for ABC to actually do anything useful for type safety. Right now the type system permits you to create a feature subtype that does not override this method.